test/jdk/java/text/Format/DecimalFormat/ToPatternTest.java

```diff
@@ -0,0 +1,126 @@
+    @ParameterizedTest
+    @MethodSource("minMaxDigits")
+    public void basicTest(int maxInt, int minInt, int maxFrac, int minFrac) {
+        // Use a US dFmt, which uses '.' as the decimal separator
+        DecimalFormat dFmt = (DecimalFormat) NumberFormat.getInstance(Locale.US);
+
+        dFmt.setMaximumIntegerDigits(maxInt);
+        dFmt.setMinimumIntegerDigits(minInt);
+        dFmt.setMaximumFractionDigits(maxFrac);
+        dFmt.setMinimumFractionDigits(minFrac);
+
+        String[] patterns = dFmt.toPattern().split("\\.");
+        assertEquals(2, patterns.length,
+                dFmt.toPattern() + " should be split into an integer/fraction portion");
+        String integerPattern = patterns[0];
+        String fractionPattern = patterns[1];
+
+        // Count # and 0 explicitly (since there are grouping symbols)
+        assertEquals(integerPattern.chars().filter(ch -> ch == '0').count() +
+                integerPattern.chars().filter(ch -> ch == '#').count(),
+                Math.max(dFmt.getGroupingSize(), dFmt.getMinimumIntegerDigits()) + 1);
+        assertEquals(integerPattern.chars().filter(ch -> ch == '0').count(), dFmt.getMinimumIntegerDigits());
+        assertEquals(fractionPattern.length(), dFmt.getMaximumFractionDigits());
+        assertEquals(fractionPattern.chars().filter(ch -> ch == '0').count(), dFmt.getMinimumFractionDigits());
+    }
+
+    // General and edge cases for the min and max Integer/Fraction digits
+    private static Stream<Arguments> minMaxDigits() {
+        return Stream.of(
+                Arguments.of(10, 5, 10, 5),
+                Arguments.of(0, 0, 1, 1),
+                Arguments.of(1, 1, 1, 1),
+                Arguments.of(5, 5, 5, 5),
+                Arguments.of(5, 10, 5, 10),
+                Arguments.of(333, 27, 409, 3)
+        );
+    }
+
+    // Ensure that a NegativePattern is explicitly produced when required.
+    @Test
+    public void negativeSubPatternTest() {
+        DecimalFormat dFmt = new DecimalFormat();
+        dFmt.setPositivePrefix("foo");
+        dFmt.setPositiveSuffix("bar");
+        dFmt.setNegativePrefix("baz");
+        dFmt.setNegativeSuffix("qux");
+
+        String[] patterns = dFmt.toPattern().split(";");
+        assertEquals(2, patterns.length,
+                "There should be a positivePattern and negativePattern");
+        String positivePattern = patterns[0];
+        String negativePattern = patterns[1];
+
+        assertTrue(positivePattern.startsWith(dFmt.getPositivePrefix()));
+        assertTrue(positivePattern.endsWith(dFmt.getPositiveSuffix()));
+        assertTrue(negativePattern.startsWith(dFmt.getNegativePrefix()));
+        assertTrue(negativePattern.endsWith(dFmt.getNegativeSuffix()));
+    }
+
+    // 8326908: Verify that an empty pattern DecimalFormat does not throw an
+    // OutOfMemoryError when toPattern() is invoked. Behavioral change of
+    // MAXIMUM_INTEGER_DIGITS replaced with DOUBLE_FRACTION_DIGITS for empty
```

Comment:
Should we explicitly check `new DecimalFormat("").getMaximumFractionDigits() == DOUBLE_FRACTION_DIGITS`?

Reply:
Thanks for the review, added an explicit check